core/domain/state_domain.py

```diff
@@ -918,8 +942,933 @@ def get_all_outcomes(self) -> List[Outcome]:
+                    # they are off by one value.
+                    item_to_layer_idx = {}
+                    for layer_idx, layer in enumerate(rule_spec.inputs['x']):
+                        for item in layer:
+                            item_to_layer_idx[item] = layer_idx
+
+                    for ele in equal_ordering_one_at_incorec_posn:
+                        is_off_by_one_value = False
+                        for layer_idx, layer in enumerate(ele):
+                            for item in layer:
+                                if layer_idx != item_to_layer_idx[item]:
+                                    is_off_by_one_value = True
+                        if is_off_by_one_value:
+                            raise utils.ValidationError(
+                                f'Rule - {rule_spec_index} of answer '
+                                f'group {ans_group_index} will never '
+                                f'be match because it is made '
+                                f'redundant by the IsEqualToOrdering'
+                                f'WithOneItemAtIncorrectPosition '
+                                f'rule above.'
+                            )
+
+        choices = self.customization_args['choices'].value
+        if len(choices) < 2:
+            raise utils.ValidationError(
+                'Atleast 2 choices should be there '
+                'in DragAndDropSortInput interaction.'
+            )
+
+        self._validate_choices_should_be_unique_and_empty()
+
+    def _validate_text_input(self) -> None:
+        """Validates the TextInput interaction.
+
+        Raises:
+            ValidationError. Text input height is not >= 1 and <= 10.
+            ValidationError. Duplicate rules are present.
+            ValidationError. The 'Contains' rule comes before another 'Contains'
+                rule, where 'Contains' rule string is a substring of other
+                rules string.
+            ValidationError. The 'Contains' rule comes before 'StartsWith'
+                rule, where 'Contains' rule string is a substring of other
+                rules string.
+            ValidationError. The 'Contains' rule comes before 'Equals'
+                rule, where 'Contains' rule string is a substring of other
+                rules string.
+            ValidationError. The 'StartsWith' rule comes before the 'Equals'
+                rule where the 'StartsWith' rule string is a prefix of other
+                rules string.
+            ValidationError. The 'StartsWith' rule comes before the another
+                'StartsWith' rule where the 'StartsWith' rule string is
+                a prefix of other rules string.
+        """
+        rule_spec_till_now: List[RuleSpecDict] = []
+        seen_strings_contains: List[List[str]] = []
+        seen_strings_startswith: List[List[str]] = []
+
+        # Text input height should be >= 1 and <= 10.
+        rows_value = self.customization_args['rows'].value
```

Comment:
Can be added via schema!

Reply:
Already have validation in the interaction/TextInput file

Reply:
So we can remove it from here?

Reply:
Yes! I will do that

Reply:
Done